core/templates/expressions/expression-syntax-tree.service.ts

```diff
@@ -16,417 +16,447 @@
       },
-      'log': {
-        eval: function(args) {
-          verifyNumArgs(args, 2);
-          var numericArgs = _coerceAllArgsToNumber(args);
-          var preciseAns = Math.log(numericArgs[0]) / Math.log(numericArgs[1]);
-          // We round answers to 9 decimal places, so that we don't run into
-          // issues like log(9, 3) = 2.0000000000004.
-          return Math.round(preciseAns * Math.pow(10, 9)) / Math.pow(10, 9);
-        },
-        getType: function(args) {
-          verifyNumArgs(args, 2);
-          _verifyArgTypesMatchExpectedType(args, PARAMETER_TYPES.REAL);
-          return PARAMETER_TYPES.REAL;
-        }
+      getType: (args: Array<string>): string => {
+        this.verifyNumArgs(args, 2);
+        this._verifyArgTypesMatchExpectedType(args,
+          AppConstants.PARAMETER_TYPES.REAL);
+        return AppConstants.PARAMETER_TYPES.REAL;
+      }
+    },
+    log: {
+      eval: (args: Array<string>): number => {
+        this.verifyNumArgs(args, 2);
+        var numericArgs = this._coerceAllArgsToNumber(args);
+        var preciseAns = Math.log(numericArgs[0]) / Math.log(numericArgs[1]);
+        // We round answers to 9 decimal places, so that we don't run into
+        // issues like log(9, 3) = 2.0000000000004.
+        return Math.round(preciseAns * Math.pow(10, 9)) / Math.pow(10, 9);
       },
-      'abs': {
-        eval: function(args) {
-          verifyNumArgs(args, 1);
-          var numericArgs = _coerceAllArgsToNumber(args);
-          return Math.abs(numericArgs[0]);
-        },
-        getType: function(args) {
-          verifyNumArgs(args, 1);
-          _verifyArgTypesMatchExpectedType(args, PARAMETER_TYPES.REAL);
-          return PARAMETER_TYPES.REAL;
-        }
+      getType: (args: Array<string>): string => {
+        this.verifyNumArgs(args, 2);
+        this._verifyArgTypesMatchExpectedType(args,
+          AppConstants.PARAMETER_TYPES.REAL);
+        return AppConstants.PARAMETER_TYPES.REAL;
       }
-      /* eslint-enable quote-props */
-    };
+    },
+    abs: {
+      eval: (args: Array<string>): number => {
+        this.verifyNumArgs(args, 1);
+        var numericArgs = this._coerceAllArgsToNumber(args);
+        return Math.abs(numericArgs[0]);
+      },
+      getType: (args: Array<string>): string => {
+        this.verifyNumArgs(args, 1);
+        this._verifyArgTypesMatchExpectedType(args,
+          AppConstants.PARAMETER_TYPES.REAL);
+        return AppConstants.PARAMETER_TYPES.REAL;
+      }
+    }
+    /* eslint-enable quote-props */
+  };
 
-    return {
-      ExpressionError: ExpressionError,
-      ExprUndefinedVarError: ExprUndefinedVarError,
-      ExprWrongNumArgsError: ExprWrongNumArgsError,
-      ExprWrongArgTypeError: ExprWrongArgTypeError,
-      applyFunctionToParseTree: applyFunctionToParseTree,
-      getParamsUsedInExpression: getParamsUsedInExpression,
-      lookupEnvs: lookupEnvs
-    };
-  }
-]);
+  public ExpressionError = ExpressionError;
```

Comment:
These should no longer be necessary after declaring them as classes (see my first comment)

Reply:
@brianrodri They are used in *expression-interpolation.service.ts* and other services or tests: 
```
if ((e instanceof ExpressionParserService.SyntaxError) ||
   (e instanceof ExpressionSyntaxTreeService.ExpressionError)) {
       return EXPRESSION_ERROR_TAG;
}
```
It won't pass if I would have to remove them unless I could replace it from `ExpressionSyntaxTree.ExpressionError` to only `ExpressionError` and in several other services and tests

Reply:
Got, in that case it's fine to keep these lines as-is (we should still be able to make them classes)